v1/src/test/java/com/google/cloud/teleport/templates/PubsubToAvroLT.java

```diff
@@ -0,0 +1,215 @@
+  private static final long TIMEOUT_FOR_1_HOUR_TEST_MINUTES = 60;
+  private static final Pattern EXPECTED_PATTERN =
+      Pattern.compile(".*" + AVRO_OUTPUT_FILENAME_PREFIX + ".*");
+
+  private static PubsubResourceManager pubsubResourceManager;
+  private static ArtifactClient gcsClient;
+
+  @Before
+  public void setup() throws IOException {
+    // Set up resource managers
+    pubsubResourceManager =
+        PubsubResourceManager.builder(testName, project)
+            .credentialsProvider(CREDENTIALS_PROVIDER)
+            .build();
+
+    Storage storageClient = createStorageClient(CREDENTIALS);
+    gcsClient = GcsArtifactClient.builder(storageClient, ARTIFACT_BUCKET, TEST_ROOT_DIR).build();
+  }
+
+  @After
+  public void teardown() {
+    ResourceManagerUtils.cleanResources(pubsubResourceManager, gcsClient);
+  }
+
+  @Test
+  public void testBacklog10gb() throws IOException, ParseException, InterruptedException {
+    testBacklog(
+        Function.identity(), NUM_MESSAGES_FOR_10_GB_TEST, TIMEOUT_FOR_10_GB_TEST_IN_MINUTES);
+  }
+
+  @Test
+  public void testSteadyState1hr() throws IOException, ParseException, InterruptedException {
+    testSteadyState1hr(Function.identity());
+  }
+
+  private void testBacklog(
+      Function<PipelineLauncher.LaunchConfig.Builder, PipelineLauncher.LaunchConfig.Builder>
+          paramsAdder,
+      long numMessages,
+      long timeoutMin)
+      throws IOException, InterruptedException, ParseException {
+    TopicName backlogTopic = pubsubResourceManager.createTopic("backlog-input");
+
+    DataGenerator dataGenerator =
+        DataGenerator.builderWithSchemaTemplate(testName, "GAME_EVENT")
+            .setQPS("1000000")
+            .setMessagesLimit(String.valueOf(numMessages))
+            .setTopic(backlogTopic.toString())
+            .setNumWorkers("50")
+            .setMaxNumWorkers("100")
+            .build();
+
+    String outputDirectoryPath =
+        ArtifactUtils.getFullGcsPath(
+            ARTIFACT_BUCKET, getClass().getSimpleName(), gcsClient.runId(), testName);
+    String tempDirectoryPath =
+        ArtifactUtils.getFullGcsPath(
+            ARTIFACT_BUCKET, getClass().getSimpleName(), gcsClient.runId(), AVRO_TEMP_DIR);
+
+    PipelineLauncher.LaunchConfig options =
+        paramsAdder
+            .apply(
+                PipelineLauncher.LaunchConfig.builder(testName, SPEC_PATH)
+                    .addParameter("inputTopic", backlogTopic.toString())
+                    .addParameter("outputDirectory", outputDirectoryPath)
+                    .addParameter("avroTempDirectory", tempDirectoryPath)
+                    .addParameter("outputFilenamePrefix", AVRO_OUTPUT_FILENAME_PREFIX))
+            .build();
+
+    // Act
+    PipelineLauncher.LaunchInfo info = pipelineLauncher.launch(project, region, options);
+    assertThatPipeline(info).isRunning();
+
+    // Wait for the template to start
+    TimeUnit.SECONDS.sleep(WAIT_FOR_WORKERS_TIME_SEC);
```

Comment:
This sleep is not required, we can remove this.

Reply:
Done